### PR TITLE
Update the README to align with the documents

### PR DESCRIPTION
**Pull request by thagikura**
Basic strategies:
- Link to relevant section in documents from README
- Use the text relevant to snippets/screenshots

The current version is available at https://github.com/thagikura/user-interface-samples/blob/widget_s_update_readme/AppWidget/README.md

### CODE REVIEW

File: AppWidget/README.md [hunk 1]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
 
+  In Android 12, the widget preview displayed in the widget picker consists of a scalable preview, which you’ll provide as an XML layout set to the widget's default size. Previously, the widget preview was a static drawable resource, in some cases leading to previews not accurately reflecting widgets after they were added to the home screen.
+
+  To implement scalable widget previews, use the [previewLayout](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo?db=dac-team#previewLayout) attribute of the appwidget-provider element to provide an XML layout instead:
+
+  [`app_widget_info_checkbox_list`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L50)
   ```xml
-  <appwidget-provider
-        android:description="@string/app_widget_grocery_list_description"
+    <appwidget-provider
```
Comment (chrisbanes) [inline AppWidget/README.md:23]: Nit: why is this indented?

File: AppWidget/README.md [hunk 2]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
-  static drawable resource using `previewImage` attribute.
-  Note that the preview may not reflect the widget after it's actually
-  placed (e.g. colors from themes)
+  Provide a description for your widget using the description attribute of appwidget-provider:
 
+  [`app_widget_info_checkbox_list`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L40)
   ```xml
-    <appwidget-provider
-        android:previewLayout="@layout/widget_grocery_list"
+  <appwidget-provider
```
Comment (chrisbanes) [inline AppWidget/README.md:36]: Ditto, why indented?

File: AppWidget/README.md [hunk 3]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
+      alt="screenshot for dynamic coloring on light theme"
+      title="screenshot for dynamic coloring on light theme" />
+  <img src="screenshots/dynamic_coloring_dark_theme.png" width="240px"
+      alt="screenshot for dynamic coloring on dark theme"
+      title="screenshot for dynamic coloring on dark theme" />
 
-  In API level 31, the colors of your widget are dynamically determined by the system colors.
-  You should use the system's default theme in order to apply the dynamic coloring (`Theme.DeviceDefault` and
-  `Theme.DeviceDefault.DayNight` for v31).
+## [Backward-compatibility with dynamic colors](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#backward-compatibility_with_dynamic_colors)
```
Comment (chrisbanes) [inline AppWidget/README.md:57]:
```suggestion
## [Backward-compatibility with dynamic colors](https://developer.android.com/about/versions/12/features/widgets#backward-compatibility_with_dynamic_colors)
```

File: AppWidget/README.md [hunk 4]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
+  
+  We recommend creating a custom theme and overriding it for Android 12. The following examples show how to do this with various types of XML files:
```
Comment (chrisbanes) [inline AppWidget/README.md:59]:
```suggestion
  We recommend creating a custom theme and overriding it when running on devices with Android 12+. The following examples show how to do this with various types of XML files:
```

File: AppWidget/README.md [hunk 5]
```diff
@@ -73,26 +85,15 @@ List of API changes
   </LinearLayout>
   ```
 
-  <img src="screenshots/dynamic_coloring_light_theme.png" width="240px"
-      alt="screenshot for dynamic coloring on light theme"
-      title="screenshot for dynamic coloring on light theme" />
-  <img src="screenshots/dynamic_coloring_dark_theme.png" width="240px"
-      alt="screenshot for dynamic coloring on dark theme"
-      title="screenshot for dynamic coloring on dark theme" />
+# [Implement rounded corners](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#rounded-corner)
```
Comment (chrisbanes) [inline AppWidget/README.md:88]:
```suggestion
# [Implement rounded corners](https://developer.android.com/about/versions/12/features/widgets#rounded-corner)
```

File: AppWidget/README.md [hunk 6]
```diff
@@ -142,28 +173,33 @@ List of API changes
...
+
+  [`xml/app_widget_info_checkbox_list.xml`](app/src/main/res/xml/app_widget_info_checkbox_list.xml\#L39)
   ```xml
   <appwidget-provider
       android:configure="com.example.android.appwidget.ListWidgetConfigureActivity"
       android:widgetFeatures="reconfigurable"
       ... />
   ```
+  
+## [Use the widget's default configuration](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#use-default)
```
Comment (chrisbanes) [inline AppWidget/README.md:200]:
```suggestion
## [Use the widget's default configuration](https://developer.android.com/about/versions/12/features/widgets#use-default)
```

File: AppWidget/README.md [hunk 7]
```diff
@@ -142,28 +173,33 @@ List of API changes
...
+  If you specify a configuration activity with the [configure](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo?db=dac-team#configure) attribute of [appwidget-provider](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo?db=dac-team), the App Widget host launches that activity immediately after a user adds the widget to their home screen.
 
-  In API level 31, users are able to reconfigure widgets after they are added to the home screen
-  by long pressing on the widget and clicking the reconfigure button(the button visible at the
-  bottom right corner after long pressing the widget).
-  You need to specify `reconfigurable` value for the `widgetFeatures` attribute.
+  Android 12 adds new options to let you provide a better configuration experience for users.
 
-  [`xml/app_widget_info_grocery_list.xml`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L39)
+## [Enable users to reconfigure placed widgets](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#reconfigure-widgets)
```
Comment (chrisbanes) [inline AppWidget/README.md:182]:
```suggestion
## [Enable users to reconfigure placed widgets](https://developer.android.com/about/versions/12/features/widgets#reconfigure-widgets)
```

File: AppWidget/README.md [hunk 8]
```diff
@@ -142,28 +173,33 @@ List of API changes
       alt="screenshot for a widget with rounded corners"
       title="screenshot for a widget with rounded corners" />
 
-- **Deferrable configurability**
+# [Make it easier to personalize widgets](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#personalize-widgets)
```
Comment (chrisbanes) [inline AppWidget/README.md:176]:
```suggestion
# [Make it easier to personalize widgets](https://developer.android.com/about/versions/12/features/widgets#personalize-widgets)
```

File: AppWidget/README.md [hunk 9]
```diff
@@ -73,26 +85,15 @@ List of API changes
...
-  ```
-  system_app_widget_background_radius: The corner radius of the widget background
-  system_app_widget_inner_radius: The corner radius of inner containers
-  ```
+  - `system_app_widget_background_radius`: The corner radius of the widget background, which will never be larger than 28dp.
+  - `system_app_widget_inner_radius`: The corner radius of any view inside the widget. This is exactly 8dp less than the background radius to align nicely when using an 8dp padding.
 
-  The recommended approach is to define custom attributes to match those system parameters
-  below the API level 31 to provide backward compatibility.
+## [Backward-compatibility with rounded corners](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#backward-compatibility_with_rounded_corners)
```
Comment (chrisbanes) [inline AppWidget/README.md:95]:
```suggestion
## [Backward-compatibility with rounded corners](https://developer.android.com/about/versions/12/features/widgets#backward-compatibility_with_rounded_corners)
```

File: AppWidget/README.md [hunk 10]
```diff
@@ -249,24 +290,26 @@ List of API changes
...
 
-- **Specifying size at compile time**
+  Starting in Android 12, you can provide more refined size attributes and more flexible layouts and by doing the following:
 
-  In API level 31, following new attributes are introduced to ensure more reliable sizes for the widget across devices and size permutations.
+  1. [Specify additional widget sizing constraints](#specify-additional-widget-sizing-constraints)
+  2. [Provide responsive layouts](#provide-responsive-layouts) or [exact layouts](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#provide-exact-layouts)
 
-  - `targetCellWidth/targetCellHeight`
+## [Specify additional widget sizing constraints](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#specify-widget-size-constraints)
```
Comment (chrisbanes) [inline AppWidget/README.md:300]:
```suggestion
## [Specify additional widget sizing constraints](https://developer.android.com/about/versions/12/features/widgets#specify-widget-size-constraints)
```

File: AppWidget/README.md [hunk 11]
```diff
@@ -249,24 +290,26 @@ List of API changes
...
-  `setViewTypeCount` should be used if you have some RemoteViews items with layouts that are sometimes present and sometimes aren’t.
-  If you just have one item type or you have some headers and then a variable number of items, you can ignore this.
+# [Use improved APIs for widget sizes and layouts](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#refine-widget-sizes)
 
-- **Specifying size at compile time**
+  Starting in Android 12, you can provide more refined size attributes and more flexible layouts and by doing the following:
 
-  In API level 31, following new attributes are introduced to ensure more reliable sizes for the widget across devices and size permutations.
+  1. [Specify additional widget sizing constraints](#specify-additional-widget-sizing-constraints)
+  2. [Provide responsive layouts](#provide-responsive-layouts) or [exact layouts](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#provide-exact-layouts)
```
Comment (chrisbanes) [inline AppWidget/README.md:298]:
```suggestion
  2. [Provide responsive layouts](#provide-responsive-layouts) or [exact layouts](https://developer.android.com/about/versions/12/features/widgets#provide-exact-layouts)
```

File: AppWidget/README.md [hunk 12]
```diff
@@ -284,9 +327,13 @@ List of API changes
        alt="Screenshot of a widget being resized"
        title="Screenshot of a widget being resized" />
 
-- **Flexible layouts**
+## [Provide responsive layouts](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#provide-responsive-layouts)
```
Comment (chrisbanes) [inline AppWidget/README.md:330]:
```suggestion
## [Provide responsive layouts](https://developer.android.com/about/versions/12/features/widgets#provide-responsive-layouts)
```

File: AppWidget/README.md [hunk 13]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
 List of API changes
 ------------
 
-- **Widget description**
+# [Improve your app's widget picker experience](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#improve-widget-picker-experience)
 
-  In API level 31, a widget description is shown for each widget in the widget
-  picker if [`description`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L40) attribute is provided for your `appwidget-provider`
+## [Add scalable widget previews to the widget picker](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#add-scalable-widget-previews)
```
Comment (chrisbanes) [inline AppWidget/README.md:15]:
```suggestion
## [Add scalable widget previews to the widget picker](https://developer.android.com/about/versions/12/features/widgets#add-scalable-widget-previews)
```

File: AppWidget/README.md [hunk 14]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
 List of API changes
 ------------
 
-- **Widget description**
+# [Improve your app's widget picker experience](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#improve-widget-picker-experience)
```
Comment (chrisbanes) [inline AppWidget/README.md:13]:
```suggestion
# [Improve your app's widget picker experience](https://developer.android.com/about/versions/12/features/widgets#improve-widget-picker-experience)
```

File: AppWidget/README.md [hunk 15]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
   ```xml
-  <appwidget-provider
-        android:description="@string/app_widget_grocery_list_description"
+    <appwidget-provider
+        android:previewLayout="@layout/widget_grocery_list"
     ... />
   ```
 
-- **Preview layout**
+## [Add a description for your widget](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#add-widget-description)
```
Comment (chrisbanes) [inline AppWidget/README.md:28]:
```suggestion
## [Add a description for your widget](https://developer.android.com/about/versions/12/features/widgets#add-widget-description)
```

File: AppWidget/README.md [hunk 16]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
   ```
-
+   
   <img src="screenshots/widget_preview.png"
     alt="Image of the widget with description and previewLayout"
     title="Image of the widget with description and previewLayout"
     />
 
-- **Dynamic coloring**
+# [Apply dynamic colors](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#dynamic-colors)
```
Comment (chrisbanes) [inline AppWidget/README.md:46]:
```suggestion
# [Apply dynamic colors](https://developer.android.com/about/versions/12/features/widgets#dynamic-colors)
```

File: AppWidget/README.md [hunk 17]
```diff
@@ -249,24 +290,26 @@ List of API changes
   )
   ```
 
-  `setViewTypeCount` should be used if you have some RemoteViews items with layouts that are sometimes present and sometimes aren’t.
-  If you just have one item type or you have some headers and then a variable number of items, you can ignore this.
+# [Use improved APIs for widget sizes and layouts](https://developer.android.com/about/versions/12/features/widgets?db=dac-team#refine-widget-sizes)
```
Comment (chrisbanes) [inline AppWidget/README.md:293]:
```suggestion
# [Use improved APIs for widget sizes and layouts](https://developer.android.com/about/versions/12/features/widgets#refine-widget-sizes)
```

File: AppWidget/README.md (hunk 14 above)
Comment (thagikura) [reply AppWidget/README.md:13]: Done

File: AppWidget/README.md (hunk 13 above)
Comment (thagikura) [reply AppWidget/README.md:15]: Done.

File: AppWidget/README.md (hunk 1 above)
Comment (thagikura) [reply AppWidget/README.md:23]: It wasn't intended. Removed the indentation

File: AppWidget/README.md (hunk 15 above)
Comment (thagikura) [reply AppWidget/README.md:28]: Done.

File: AppWidget/README.md (hunk 2 above)
Comment (thagikura) [reply AppWidget/README.md:36]: It wasn't intended. Removed the indentation.

File: AppWidget/README.md (hunk 16 above)
Comment (thagikura) [reply AppWidget/README.md:46]: Done.

File: AppWidget/README.md (hunk 3 above)
Comment (thagikura) [reply AppWidget/README.md:57]: Done.

File: AppWidget/README.md (hunk 4 above)
Comment (thagikura) [reply AppWidget/README.md:59]: Done.

File: AppWidget/README.md (hunk 5 above)
Comment (thagikura) [reply AppWidget/README.md:88]: Done.

File: AppWidget/README.md (hunk 9 above)
Comment (thagikura) [reply AppWidget/README.md:95]: Done.

File: AppWidget/README.md (hunk 8 above)
Comment (thagikura) [reply AppWidget/README.md:176]: Done.

File: AppWidget/README.md (hunk 7 above)
Comment (thagikura) [reply AppWidget/README.md:182]: Done.

File: AppWidget/README.md (hunk 6 above)
Comment (thagikura) [reply AppWidget/README.md:200]: Done.

File: AppWidget/README.md (hunk 17 above)
Comment (thagikura) [reply AppWidget/README.md:293]: Done.

File: AppWidget/README.md (hunk 11 above)
Comment (thagikura) [reply AppWidget/README.md:298]: Done.

File: AppWidget/README.md (hunk 10 above)
Comment (thagikura) [reply AppWidget/README.md:300]: Done.

File: AppWidget/README.md (hunk 12 above)
Comment (thagikura) [reply AppWidget/README.md:330]: Done.

**Comment by thagikura** [comment]
Thanks @chrisbanes addressed your feedback.

**Review by chrisbanes — APPROVED** [review]

**Comment by yogishadixit** [comment]
I worry that we have too much duplicated text between the DAC and the README. This will make it more challenging to keep both in sync in the future. 

I think it should be sufficient to have just the DAC link and the code link in each section. For example:

**[Add scalable widget previews to the widget picker](https://developer.android.com/about/versions/12/features/widgets#add-scalable-widget-previews)**
Use the previewLayout attribute of the appwidget-provider element to provide an XML layout. See [app_widget_info_checkbox_list.xml](app_widget_info_checkbox_list).

**[Apply dynamic colors](https://developer.android.com/about/versions/12/features/widgets#dynamic-colors)**
Create a custom theme and override it with device default values in Android 12+. For example, see [values/themes.xml](values/themes.xml) and [values-v31/themes.xml](values-v31/themes.xml)

File: AppWidget/README.md [hunk 18]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
-- **Widget description**
+# [Improve your app's widget picker experience](https://developer.android.com/about/versions/12/features/widgets#improve-widget-picker-experience)
 
-  In API level 31, a widget description is shown for each widget in the widget
-  picker if [`description`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L40) attribute is provided for your `appwidget-provider`
+## [Add scalable widget previews to the widget picker](https://developer.android.com/about/versions/12/features/widgets#add-scalable-widget-previews)
 
+  In Android 12, the widget preview displayed in the widget picker consists of a scalable preview, which you’ll provide as an XML layout set to the widget's default size. Previously, the widget preview was a static drawable resource, in some cases leading to previews not accurately reflecting widgets after they were added to the home screen.
+
+  To implement scalable widget previews, use the [previewLayout](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#previewLayout) attribute of the appwidget-provider element to provide an XML layout instead:
```
Comment (michaelmauzy) [inline AppWidget/README.md:19]: `previewLayout`

File: AppWidget/README.md [hunk 19]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
+## [Add a description for your widget](https://developer.android.com/about/versions/12/features/widgets#add-widget-description)
 
-  The widget preview is displayed from the layout XML if it's provided as
-  [`previewLayout`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L50) attribute for your `appwidget-provider` instead of a
-  static drawable resource using `previewImage` attribute.
-  Note that the preview may not reflect the widget after it's actually
-  placed (e.g. colors from themes)
+  In Android 12, you can optionally provide a description for the widget picker to display for your widget. 
 
+  Provide a description for your widget using the description attribute of appwidget-provider:
```
Comment (michaelmauzy) [inline AppWidget/README.md:32]: `appwidget-provider`

File: AppWidget/README.md [hunk 20]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
-    ... />
+  <appwidget-provider
+      android:description="@string/app_widget_grocery_list_description"
+      ... />
   ```
-
+   
   <img src="screenshots/widget_preview.png"
     alt="Image of the widget with description and previewLayout"
     title="Image of the widget with description and previewLayout"
```
Comment (michaelmauzy) [inline AppWidget/README.md:43]: `previewLayout`

File: AppWidget/README.md [hunk 21]
```diff
@@ -10,41 +10,53 @@ The initial set of new APIs focuses on improving existing RemoteView APIs.
...
+  <img src="screenshots/dynamic_coloring_light_theme.png" width="240px"
+      alt="screenshot for dynamic coloring on light theme"
+      title="screenshot for dynamic coloring on light theme" />
+  <img src="screenshots/dynamic_coloring_dark_theme.png" width="240px"
+      alt="screenshot for dynamic coloring on dark theme"
+      title="screenshot for dynamic coloring on dark theme" />
+
+## [Backward-compatibility with dynamic colors](https://developer.android.com/about/versions/12/features/widgets#backward-compatibility_with_dynamic_colors)
+  
+  We recommend creating a custom theme and overriding it when running on devices with Android 12+. The following examples show how to do this with various types of XML files:
```
Comment (michaelmauzy) [inline AppWidget/README.md:59]: ... when running on devices with Android 12 or later.

File: AppWidget/README.md [hunk 22]
```diff
@@ -142,28 +173,33 @@ List of API changes
       alt="screenshot for a widget with rounded corners"
       title="screenshot for a widget with rounded corners" />
 
-- **Deferrable configurability**
+# [Make it easier to personalize widgets](https://developer.android.com/about/versions/12/features/widgets#personalize-widgets)
+
+  If you specify a configuration activity with the [configure](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#configure) attribute of [appwidget-provider](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo), the App Widget host launches that activity immediately after a user adds the widget to their home screen.
```
Comment (michaelmauzy) [inline AppWidget/README.md:178]: Enclose configure and appwidget-provider in backticks

File: AppWidget/README.md [hunk 23]
```diff
@@ -142,28 +173,33 @@ List of API changes
...
-  In API level 31, users are able to reconfigure widgets after they are added to the home screen
-  by long pressing on the widget and clicking the reconfigure button(the button visible at the
-  bottom right corner after long pressing the widget).
-  You need to specify `reconfigurable` value for the `widgetFeatures` attribute.
+  Android 12 adds new options to let you provide a better configuration experience for users.
 
-  [`xml/app_widget_info_grocery_list.xml`](app/src/main/res/xml/app_widget_info_grocery_list.xml\#L39)
+## [Enable users to reconfigure placed widgets](https://developer.android.com/about/versions/12/features/widgets#reconfigure-widgets)
+
+  To configure widgets that are labeled as reconfigurable, users can long-press the widget. This displays a Reconfigure button, which they can tap to change settings.
```
Comment (michaelmauzy) [inline AppWidget/README.md:184]: Boldface Reconfigure

File: AppWidget/README.md [hunk 24]
```diff
@@ -142,28 +173,33 @@ List of API changes
...
+
+  <img src="screenshots/widget_reconfigure.png"
+      alt="screenshot for a widget with reconfigure button"
+      title="screenshot for a widget with reconfigure button" />
+
+  Specify the `reconfigurable` flag in the `widgetFeatures` attribute of appwidget-provider:
```
Comment (michaelmauzy) [inline AppWidget/README.md:190]: `appwidget-provider`

File: AppWidget/README.md [hunk 25]
```diff
@@ -172,25 +208,30 @@ List of API changes
...
   ```
 
-- **Transitions**
+# [Enable smoother transitions](https://developer.android.com/about/versions/12/features/widgets#enable-smoother-transitions)
 
-  In API level 31, there is a smooth transition from the widget to the app when the user launches the app from the widget.
+  In Android 12, launchers provide a smoother transition when a user launches your app from a widget.
 
-  You need to specify the id of the background element of the widget with the [android:id/background](https://developer.android.com/reference/android/R.id#background) to enable this smooth transition.
+  To enable this improved transition, use [android:id/background](https://developer.android.com/reference/android/R.id#background)  or android.R.id.background to identify your background element:
```
Comment (michaelmauzy) [inline AppWidget/README.md:215]: `android:id/background`

`android.R.id.background`

File: AppWidget/README.md [hunk 26]
```diff
@@ -172,25 +208,30 @@ List of API changes
...
 
-- **Compound buttons**
+# [Add new compound buttons](https://developer.android.com/about/versions/12/features/widgets#new-compound-buttons)
+
+  Android 12 adds new support for stateful behavior using the following existing components:
 
-  In API level 31, therer are three new compound buttons to use in a widget to support stateful behavior.
+  - [Checkbox](https://developer.android.com/reference/kotlin/android/widget/CheckBox)
+  - [Switch](https://developer.android.com/reference/kotlin/android/widget/Switch)
+  - [RadioButton](https://developer.android.com/reference/kotlin/android/widget/RadioButton)
```
Comment (michaelmauzy) [inline AppWidget/README.md:234]: `Checkbox`
`Switch`
`RadioButton`

File: AppWidget/README.md [hunk 27]
```diff
@@ -227,14 +268,14 @@ List of API changes
   ```
   See the documents for [fromPendingIntent](https://developer.android.com/reference/android/widget/RemoteViews.RemoteResponse#fromPendingIntent(android.app.PendingIntent)) and [fromFillInIntent](https://developer.android.com/reference/android/widget/RemoteViews.RemoteResponse#fromFillInIntent(android.content.Intent)) for more details.
```
Comment (michaelmauzy) [inline AppWidget/README.md:269]: `fromPendingIntent`

`fromFillInIntent`

File: AppWidget/README.md [hunk 28]
```diff
@@ -227,14 +268,14 @@ List of API changes
...
 
+# [Use simplified RemoteViews collections](https://developer.android.com/about/versions/12/features/widgets#leverage-simplified-remoteview-collections)
 
-- **Simplified RemoteView collections**
- 
-  When you use a collection of RemoteViews (ListView) you need to declare and write a [RemoteViewService](https://developer.android.com/reference/android/widget/RemoteViewsService) to return 
-  [RemoteViewsFactory](https://developer.android.com/reference/android/widget/RemoteViewsService.RemoteViewsFactory), which complicates the code in AppWidgetProvider and forces the system to bind and start up your service
-  even if the collection has only few items.
+  Android 12 adds the `setRemoteAdapter(int viewId, RemoteViews.RemoteCollectionItems items)` method, which lets your app pass along a collection directly when populating a ListView. Previously, when using a ListView, it was necessary to implement and declare a [RemoteViewService](https://developer.android.com/reference/android/widget/RemoteViewsService) to return
```
Comment (michaelmauzy) [inline AppWidget/README.md:273]: `ListView`

File: AppWidget/README.md [hunk 29]
```diff
@@ -227,14 +268,14 @@ List of API changes
...
-  [RemoteViewsFactory](https://developer.android.com/reference/android/widget/RemoteViewsService.RemoteViewsFactory), which complicates the code in AppWidgetProvider and forces the system to bind and start up your service
-  even if the collection has only few items.
+  Android 12 adds the `setRemoteAdapter(int viewId, RemoteViews.RemoteCollectionItems items)` method, which lets your app pass along a collection directly when populating a ListView. Previously, when using a ListView, it was necessary to implement and declare a [RemoteViewService](https://developer.android.com/reference/android/widget/RemoteViewsService) to return 
+  [RemoteViewsFactory](https://developer.android.com/reference/android/widget/RemoteViewsService.RemoteViewsFactory).
 
-  In API level 31, a new API allows you to pass a collection of RemoteViews directly when populating a ListView.
+  If the collection doesn’t use a constant set of layouts (in other words, if some items are only sometimes present), use `setViewTypeCount` to specify the maximum number of unique layouts the collection can contain.
+
+  Here’s an example of how to implement simplified RemoteViews collections.
```
Comment (michaelmauzy) [inline AppWidget/README.md:278]: `RemoteViews`

File: AppWidget/README.md [hunk 30]
```diff
@@ -249,24 +290,26 @@ List of API changes
...
 
-  - `targetCellWidth/targetCellHeight`
+## [Specify additional widget sizing constraints](https://developer.android.com/about/versions/12/features/widgets#specify-widget-size-constraints)
 
-    Defines the default size of the widget measured in cells of the launcher home. If they are defined, those values will be used instead of `minWidth/minHeight` (that are available in API level < 31).
+  Android 12 adds new APIs allowing you to ensure your widget is sized more reliably across different devices with varying screen sizes.
 
-  - `maxResizeWidth/maxResizeHeight`
+  In addition to the existing [minWidth](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#minWidth), [minHeight](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#minHeight), [minResizeWidth](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#minResizeWidth), and [minResizeHeight](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#minResizeHeight) attributes, use the following new appwidget-provider attributes:
```
Comment (michaelmauzy) [inline AppWidget/README.md:304]: Put backticks around all APIs in this paragraph.

File: AppWidget/README.md [hunk 31]
```diff
@@ -249,24 +290,26 @@ List of API changes
...
 
-    Defines the maximum size of the widget in the launcher home.
+  - [targetCellWidth](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#targetCellWidth) and [targetCellHeight](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#targetCellHeight): Defines the target size of the widget in terms of launcher grid cells. If defined, these attributes are used instead of minWidth or minHeight. 
 
-  It's recommended to use the new attributes above in addition to existing `minWidth/minHeight` and `minResizeWidth/minResizeHeight` attributes.
+  - [maxResizeWidth](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeWidth) and [maxResizeHeight](https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#maxResizeHeight): Defines the maximum size the launcher allows the user to resize the widget.
```
Comment (michaelmauzy) [inline AppWidget/README.md:308]: `maxResizeWidth`

**Review by michaelmauzy — COMMENTED** [review]
Comments posted.

**Comment by chrisbanes** [comment]
Manually merged this in #244 (I couldn't push to your branch @thagikura). Closing this.

**Pull request closed without merging**